common/src/main/java/com/wynntils/services/counter/CounterService.java

```diff
@@ -0,0 +1,84 @@
+package com.wynntils.services.counter;
+
+import com.wynntils.core.components.Service;
+import com.wynntils.mc.event.TickEvent;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class CounterService extends Service {
+    private final Map<String, Counter> counterMap = new HashMap<>();
+
+    public CounterService() {
+        super(List.of());
+    }
+
+    @SubscribeEvent
+    public void onTick(TickEvent e) {
+        counterMap.forEach((id, counter) -> {
+            if (counter.getCurrentTicks() >= counter.getMaxTicks()) {
+                if (counter.getCurrent() >= counter.getMax()) {
+                    counter.setCurrent(0);
+                } else {
+                    counter.setCurrent(counter.getCurrent() + 1);
+                }
+                counter.setCurrentTicks(0);
+            } else {
+                counter.setCurrentTicks(counter.getCurrentTicks() + 1);
+            }
+        });
+    }
+
+    public Integer getCounterValue(String id) {
+        return counterMap.get(id).getCurrent();
+    }
+
+    public boolean counterExists(String id) {
+        return counterMap.containsKey(id);
+    }
+
+    public void createCounter(Integer max, Integer maxTicks, String id) {
```

Comment:
The names are confusing. `maxTicks` means `ticksPerIncrement`. right? And `max` means `wrapAfter` (or more technically, `modulus`)?

Reply:
Yeah. I will refactor to something like https://discord.com/channels/394189072635133952/946471329857675307/1136544889551732747 when I have time. Maybe also get rid of id and replace with hash code or something like that